app/experimenter/normandy/tasks.py

```diff
@@ -165,11 +165,12 @@ def update_population_percent(experiment, recipe_data, filter_objects):
     if bucket_sample := filter_objects.get("bucketSample") or filter_objects.get(
         "namespaceSample"
     ):
-        percent_decimal = decimal.Decimal(
-            bucket_sample["count"] / bucket_sample["total"] * 100
-        )
-        if experiment.population_percent != percent_decimal:
-            changed_data = {"population_percent": percent_decimal}
+
+        percent_decimal = bucket_sample["count"] / bucket_sample["total"] * 100
+        rounded_decimal = format(percent_decimal, ".4f")
+
+        if str(experiment.population_percent) != rounded_decimal:
+            changed_data = {"population_percent": rounded_decimal}
```

Comment:
Oh you're comparing them as strings, in theory that works, I was thinking of explicitly setting the Decimal precision and comparing them that way:

https://gist.github.com/jackiekazil/6201722#3-round-decimal-by-setting-precision

I can't think off the top of my head whether that makes a real difference, but if this goes sideways then we can try it that way.